src/parsers/manifest/dash/indexes/timeline/timeline_representation_index.ts

```diff
@@ -564,27 +565,55 @@ export default class TimelineRepresentationIndex implements IRepresentationIndex
    * Call this function when the timeline is unknown.
    * This function was added to only perform that task lazily, i.e. only when
    * first needed.
+   * After calling it, every now unneeded variable will be freed from memory.
+   * This means that calling _getTimeline more than once will just return an
+   * empty array.
+   *
+   * /!\ Please note that this structure should follow the exact same structure
+   * than a SegmentTimeline element in the corresponding MPD.
+   * This means:
+   *   - It should have the same amount of elements in its array than there was
+   *     `<S>` elements in the SegmentTimeline.
+   *   - Each of those same elements should have the same start time, the same
+   *     duration and the same repeat counter than what could be deduced from
+   *     the SegmentTimeline.
+   * This is needed to be able to run parsing optimization when refreshing the
+   * MPD. Not doing so could lead to the RxPlayer not being able to play the
+   * stream anymore.
    * @returns {Array.<Object>}
    */
-  private _constructTimeline() : IIndexSegment[] {
-    const initialTimeline = this._parseTimeline();
-    const timeline : IIndexSegment[] = [];
-    for (let i = 0; i < initialTimeline.length; i++) {
-      const item = initialTimeline[i];
-      const nextItem = timeline[timeline.length - 1] === undefined ?
-        null :
-        timeline[timeline.length - 1];
-      const prevItem = initialTimeline[i + 1] === undefined ?
-        null :
-        initialTimeline[i + 1];
-      const timelineElement = fromParsedSToIndexSegment(item,
-                                                        nextItem,
-                                                        prevItem,
-                                                        this._scaledPeriodStart);
-      if (timelineElement != null) {
-        timeline.push(timelineElement);
+  private _getTimeline() : IIndexSegment[] {
+    if (this._parseTimeline === null) {
+      if (this._index.timeline !== null) {
+        return this._index.timeline;
       }
+      log.error("DASH: Timeline already lazily parsed.");
+      return [];
     }
-    return timeline;
+
+    const newElements = this._parseTimeline();
+    this._parseTimeline = null; // Free memory
+
+    if (this._unsafelyBaseOnPreviousIndex === null ||
+        newElements.length < MIN_DASH_S_ELEMENTS_TO_PARSE_UNSAFELY)
+    {
+      // Just completely parse the current timeline
+      return constructTimelineFromElements(newElements, this._scaledPeriodStart);
+    }
+
+    // Construct previously parsed timeline if not already done
```

Comment:
/** */ ?

Reply:
Same than for the other ones. I see it more as a code comment than as a JSDoc.